test/objectwrap.js

```diff
@@ -98,6 +106,9 @@ async function test (binding) {
     assert.strictEqual(obj.testMethodT(), 'method<>(const char*)');
     obj[clazz.kTestVoidMethodTInternal]('method<>(Symbol)');
     assert.strictEqual(obj[clazz.kTestMethodTInternal](), 'method<>(Symbol)');
+    try { clazz.prototype.testMethod('method'); } catch (e) { }
+    try { clazz.prototype.testMethodT(); } catch (e) { }
+    try { clazz.prototype.testVoidMethodT('method<>(const char*)'); } catch (e) { }
```

Comment:
V8 actually protects against these with "Invalid invocation". IOW, it does not allow one to invoke a native instance method where the JavaScript `this` is not an instance of the class. Thus, these checks are unnecessary. Interesting though that it allows property accesses to go through 🤨